service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/XPathService.java

```diff
@@ -53,4 +58,13 @@ public String getNodeValue(Document xmlDoc, String firstExpression, String secon
             return getNodeValue(xmlDoc, secondExpression);
         }
     }
+
+    @SneakyThrows
+    public NodeList getNodes(Document document, String xPath) {
+        XPathExpression xPathExpression = XPathFactory.newInstance()
+            .newXPath()
+            .compile(xPath);
+
+        return ((NodeList) xPathExpression.evaluate(document, NODESET));
```

Comment:
The additional brackets are redundent
```suggestion
        return (NodeList) xPathExpression.evaluate(document, NODESET);
```